provider/common/cluster_waiter.go

```diff
@@ -173,3 +173,10 @@ func pollClusterState(clusterId string, ctx context.Context, timeout int64, clus
 
 	return object, nil
 }
+
+func ValidateTimeout(timeOut *int64, defaultTimeout int64) *int64 {
+	if timeOut == nil || *timeOut <= 0 {
```

Comment:
Would it be confusing for the customer to get the default when configured below zero? I feel like it should throw an error instead. We could add a validation directly to the attribute declaration instead of having that validation here and just handling nil -> default instead